test/zksync-tests/NonceBitmap.test.ts

```diff
@@ -0,0 +1,155 @@
+            //vm.expectRevert(InvalidNonce.selector);
+            await expect(permit2.useUnorderedNonce(spender.address, ethers.BigNumber.from(254))).to.be.reverted;
+            //vm.expectRevert(InvalidNonce.selector);
+            await expect(permit2.useUnorderedNonce(spender.address, ethers.BigNumber.from(255))).to.be.reverted;
+            await (await permit2.useUnorderedNonce(spender.address, ethers.BigNumber.from(256))).wait()
+        });
+    });
+
+    describe('Test Invalidate Non zero Word', function () {
+        it('should fail 5', async function () {
+            await (await permit2.connect(spender).invalidateUnorderedNonces(ethers.constants.One, ethers.constants.Two.pow(256).sub(ethers.constants.One))).wait();
+
+            await (await permit2.useUnorderedNonce(spender.address, ethers.BigNumber.from(0))).wait();
+            await (await permit2.useUnorderedNonce(spender.address, ethers.BigNumber.from(254))).wait();
+            await (await permit2.useUnorderedNonce(spender.address, ethers.BigNumber.from(255))).wait();
+            //vm.expectRevert(InvalidNonce.selector);
+            await expect(permit2.useUnorderedNonce(spender.address, ethers.BigNumber.from(256))).to.be.reverted;
+            //vm.expectRevert(InvalidNonce.selector);
+            await expect( permit2.useUnorderedNonce(spender.address, ethers.BigNumber.from(511))).to.be.reverted;
+            await (await permit2.useUnorderedNonce(spender.address, ethers.BigNumber.from(512))).wait();
+        });
+    });
+
+    describe('Test Using Nonce Twice Fails', function () {
+        it('should fail 6', async function () {
+            let nonce: BigNumberish = ethers.BigNumber.from(Math.floor(Math.random() * 100));
+            await (await permit2.useUnorderedNonce(spender.address, nonce)).wait();
+
+            //vm.expectRevert(InvalidNonce.selector);
+            await expect(permit2.useUnorderedNonce(spender.address, nonce)).to.be.reverted;
+        });
+    });
+
+    describe('Test Use TwoRandom Nonces', function () {
+        it('should fail 7', async function () {
+            let first: BigNumberish = ethers.BigNumber.from(Math.floor(Math.random() * 100));
+            let second: BigNumberish = ethers.BigNumber.from(Math.floor(Math.random() * 100));
+
+            await (await permit2.useUnorderedNonce(spender.address, first)).wait();
+
+            if (first == second) {
+                //vm.expectRevert(InvalidNonce.selector);
+                await expect(permit2.useUnorderedNonce(spender.address, second)).to.be.reverted;
+            } else {
+                await (await permit2.useUnorderedNonce(spender.address, second)).wait();
+            }
+        });
+    });
+
+    describe('Test Invalidate Nonces Randomly', function () {
+        it('should 8', async function () {
+            let wordPos: BigNumberish = ethers.BigNumber.from(Math.floor(Math.random() * 100));
```

Comment:
This is a number in the range [0, 100], but in the original test, it is in the range [0, 2^248-1].

Reply:
Please make sure that you put the correct range for all tests with random input.